src/components/CodeSearchResults/CodeSearchResultsTable.jsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { Icon } from '@edx/paragon';
+
+import TableContainer from '../../containers/TableContainer';
+import RemindButton from '../RemindButton';
+import RevokeButton from '../RevokeButton';
+
+import EcommerceApiService from '../../data/services/EcommerceApiService';
+
+const tableColumns = [
+  {
+    label: 'Coupon Batch',
+    key: 'couponName',
+  },
+  {
+    label: 'Code',
+    key: 'code',
+  },
+  {
+    label: 'Redeemed',
+    key: 'isRedeemed',
+  },
+  {
+    label: 'Redemption Date',
+    key: 'redemptionDate',
+  },
+  {
+    label: 'Course Title',
+    key: 'courseTitle',
+  },
+  {
+    label: 'Actions',
+    key: 'actions',
+  },
+];
+
+/**
+ * The search API response contains a record for each coupon batch a
+ * user is associated with (i.e., has unused assignments, or actual
+ * redemptions). Under each coupon batch is an array of assignments and
+ * redemptiions, that contains all the metadata associated with a redemption,
+ * if applicable. Because we want the search results UI to show a row
+ * for each unused assignment and all redeemed courses associated with a
+ * particular user, this function flattens the search results by creating
+ * a new table row for each value in `redemptions_and_assignments`.
```

Comment:
might be good for @christopappas to review this part just to make sure the api contract lines up as expected